pkg/k8s/identitybackend/identity_test.go

```diff
@@ -222,6 +222,19 @@ func TestGetIdentity(t *testing.T) {
 			addWaitGroup := sync.WaitGroup{}
 			addWaitGroup.Add(len(tc.identities))
 
+			// To avoid a race, we must created these before we start ListAndWatch, see #30873.
+			// There is no easy way of knowing when the watch is established. Specifically,
+			// 'HasSynced' does _not_ guarantee it: the fake object tracker doesn't do resource
+			// versioning and hence cannot replay events in the reflector's gap between list and
+			// watch. Ironically, therefore, waiting for HasSynced, _increases_ the likelihood of
+			// the race.
```

Comment:
This comment seems weird given the code after. There is no code calling HasSynced, why are you commenting about that?

Reply:
Maybe s/waiting/if we try to wait/ ?
The part that tripped me was that I was expecting to find the "HasSynced" somewhere.